Demo/Demo/ViewModels/CardPaymentViewModel.swift

```diff
@@ -10,14 +10,33 @@ class CardPaymentViewModel: ObservableObject, CardDelegate {
 
     private var cardClient: CardClient?
 
-    func createOrder(amount: String, selectedMerchantIntegration: MerchantIntegration, intent: String) async throws {
+    func createOrder(
+        amount: String,
+        selectedMerchantIntegration: MerchantIntegration,
+        intent: String,
+        shouldVault: Bool,
+        customerID: String? = nil
+    ) async throws {
         // might need to pass in payee as payee object or as auth header
 
         let amountRequest = Amount(currencyCode: "USD", value: amount)
         // TODO: might need to pass in payee as payee object or as auth header
+
+        var vaultPaymentSource: VaultCardPaymentSource?
+        if shouldVault {
+            var customer: CardVaultCustomer?
+            if let customerID {
+                customer = CardVaultCustomer(id: customerID)
+            }
+            let attributes = Attributes(vault: Vault(storeInVault: "ON_SUCCESS"), customer: customer)
+            let card = VaultCard(attributes: attributes)
+            vaultPaymentSource = VaultCardPaymentSource(card: card)
+        }
+
         let orderRequestParams = CreateOrderParams(
             intent: intent,
-            purchaseUnits: [PurchaseUnit(amount: amountRequest)]
+            purchaseUnits: [PurchaseUnit(amount: amountRequest)],
+            paymentSource: vaultPaymentSource
         )
```

Comment:
This goes against the design of our SDK since in our other payment method flows we never expect the merchant to use the card details in an API call that they make on their own. Only the PPCP SDK is handling any API call that involves card details.

Reply:
But there are no card details in here Sammy. Just vault option passed in. Card details are passed in in confirmPaymentSource.

Reply:
Hm gotcha. Another question - why does it matter if the merchant send specific params in their CreateOrder call for the vaulting flow, if our `vault` method doesn't accept an orderID at all? (See [`CardVaultRequest` init](https://github.com/paypal/paypal-ios/blob/8b04e5fda77152afa860ea897448479e192ff256/Sources/CardPayments/Models/CardVaultRequest.swift#L17))

Reply:
This is for vault with pay during checkout. vault method will be with vault without payment.

Reply:
Either way works for me, as long as we don't forget to document that it's the `orderID` that is the magic piece required for vault with purchase